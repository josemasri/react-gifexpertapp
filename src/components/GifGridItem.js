import React from "react";

const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <p>{title.substring(0, 10)}...</p>
    </div>
  );
};

export default GifGridItem;
