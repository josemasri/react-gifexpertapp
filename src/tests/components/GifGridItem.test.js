import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas de GifGridItem", () => {
  const title = "Gif example";
  const url =
    "https://media0.giphy.com/media/LrN9NbJNp9SWQ/giphy.gif?cid=1d3f88198f97aabe3d9caa73d8b29bc357ec8a6af0ebf31c&rid=giphy.gif";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("Debe renderizarse <GifGridItem /> de manera correcta", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debe de tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text()).toBe(`${title.substring(0, 10)}...`);
  });
  test("debe tener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
  test("debe tener animate__fadeIn", () => {
    const div = wrapper.find("div");
    expect(div.hasClass("animate__fadeIn")).toBe(true);
  });
});
