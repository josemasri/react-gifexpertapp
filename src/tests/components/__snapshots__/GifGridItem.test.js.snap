// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGridItem Debe renderizarse <GifGridItem /> de manera correcta 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Gif example"
    src="https://media0.giphy.com/media/LrN9NbJNp9SWQ/giphy.gif?cid=1d3f88198f97aabe3d9caa73d8b29bc357ec8a6af0ebf31c&rid=giphy.gif"
  />
  <p>
    Gif exampl
    ...
  </p>
</div>
`;
